department-of-veterans-affairs/caseflow
# Appeals 13049, Appeals 12712 part 2 : Update MPI NOD process, fix issue with DAS Checkbox Legacy data display

Resolves [APPEALS-13049](https://vajira.max.gov/browse/APPEALS-13049) , APPEALS-12712

### Description
- Added STMDTIME update to update_veteran_sfnod method for MPI person updates
- Added script update_vacols_stmdtime_from_caseflow.rb to retroactively update CORRES.STMDTIME with the MpiUpdatePersonEvent.completed_at value for any records that have not been updated since the SFNOD was updated from Caseflow
- Added script add_corres_for_stmdtime_testing.rb to generate data locally in both VACOLS and Caseflow for testing
- Modified judge_legacy_task previous_task.assigned_at to match decass.deassign to a  priorloc.with_attorney? instead of just getting the first priorloc.with_attorney?

### Acceptance Criteria

### Testing Plan
1. See https://github.com/department-of-veterans-affairs/caseflow/pull/17918
2. See https://github.com/department-of-veterans-affairs/caseflow/pull/17919
